front/components/vaults/VaultResourcesList.tsx

```diff
@@ -175,20 +183,26 @@ export const VaultResourcesList = ({
   onSelect,
 }: VaultResourcesListProps) => {
   const [dataSourceSearch, setDataSourceSearch] = useState<string>("");
+  const [showFolderOrWebsiteModal, setShowFolderOrWebsiteModal] =
+    useState(false);
+  const [selectedDataSourceView, setSelectedDataSourceView] =
+    useState<DataSourceViewWithConnectorType | null>(null);
+  const [showDeleteConfirmDialog, setShowDeleteConfirmDialog] = useState(false);
 
   const { dataSources, isDataSourcesLoading } = useDataSources(owner);
+  const router = useRouter();
+  const sendNotification = useContext(SendNotificationsContext);
 
   const searchBarRef = useRef<HTMLInputElement>(null);
 
   const isSystemVault = systemVault.sId === vault.sId;
   const isManaged = category === "managed";
+  const isStatic = isWebsiteOrFolder(category);
```

Comment:
i'm not sure it's worth having a method here for something that is used only once ?

Reply:
I tend to the favour having methods in `types` for this kinda things so that they can be reused by others when needed. But don't mind moving the logic here if you want?

Reply:
the method looked a little bit complex (but nice typescript fun :-) ) and I was wondering if it was really needed at this point .. but as you like